whisper/whisper05/filter.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package whisper05
+
+import (
+	"crypto/ecdsa"
+
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
```

Comment:
Annoying style comment: Please try to avoid having more than two 'groups' of imports. goimports tries to maintain two groups where first one contains all stdlib imports and the second one contains 'external' imports. But sometimes it gets confused by edits and splits them up more. In that case you need to remove the blank line yourself. Please delete the blank line between "crypto/ecdsa" and "sync".